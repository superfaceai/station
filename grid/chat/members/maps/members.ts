/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

export const getMembersTest = (
  provider: string,
  options?: { workspace?: string; recordingOptions?: RecordingProcessOptions }
): void => {
  describe(`chat/members/${provider}`, () => {
    let superface: SuperfaceTest;

    describe('GetMembers', () => {
      beforeAll(() => {
        superface = new SuperfaceTest({
          profile: 'chat/members',
          provider,
          useCase: 'GetMembers',
          testInstance: expect,
        });
      });

      it('performs correctly', async () => {
        const page1 = await superface.run(
          {
            input: {
              workspace: options?.workspace,
              limit: 1,
            },
            testName: 'page 1',
          },
          options?.recordingOptions
        );

        expect(page1.isOk).toBeTruthy();
        expect(page1).toMatchSnapshot();

        /* eslint-disable @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment */
        const cursor = page1.isOk() ? (page1.value as any).nextPage : undefined;

        if (!cursor) {
          return;
        }

        const page2 = await superface.run(
          {
            input: {
              workspace: options?.workspace,
              limit: 1,
              page: cursor,
            },
            testName: 'page 2',
          },
          options?.recordingOptions
        );

        expect(page2.isOk).toBeTruthy();
        expect(page2).toMatchSnapshot();
      });
    });
  });
};
