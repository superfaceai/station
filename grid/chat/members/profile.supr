"""
Chat Members
Retrieve members of the workspace.
"""
name = "chat/members"
version = "1.0.1"

"""
Get workspace members
Retrieve paginated list of members
"""
usecase GetMembers safe {
  input {
    """
    Workspace
    Representation of a workspace. Can be ID or name, depending on the provider capability.
    """
    workspace string!
    
    """
    Limit
    Maximum number of records to retrieve
    """
    limit number
    
    """
    Page
    Page cursor retrieved from nextPage for results pagination
    """
    page string!
  }

  result {
		"""
    Members
    List of retrieved members
    """
    members! [Member]!

    """
    Next Page
    Cursor to the next page of listing
    """
    nextPage string

    """
    Rate Limit
    Rate limit details.
    """
    rateLimit RateLimit
  }
	
  error Error

  example Successful {
    input {
      workspace = '877903817948147752',
      limit = 2
    }

    result {
      members = [
        {
          id = '1'
          username = 'user1'
          joinedAt = 1643798532134.069,
        }
        {
          id = '2'
          username = 'userBot'
          joinedAt = 1643808532134.069,
          isBot = true
        }
      ]
    }
  }

  example Failed {
    input {
      workspace = 'not-existing',
    }

    error {
      title = 'workspace_not_found'
    }
  }
}

model Member {
  """
  Id
  Unique identifier of the member
  """
  id! string!

  """
  Username
  Username identifier of the member
  """
  username! string!

  """
  Joined at
  UNIX ms timestamp (e.g. 1095379198750) when the member joined workspace
  """
  joinedAt number

  """
  Is Bot
  Boolean that represents whether the member is a bot
  """
  isBot boolean
}

model Error {
  """
  Title
  A short, human-readable summary of the problem type.
  """
  title!

  """
  Detail
  A human-readable explanation specific to this occurrence of the problem.
  """
  detail

  """
  Rate Limit
  Rate limit details.
  """
  rateLimit RateLimit
}

model RateLimit {
  """
  Bucket
  Different parts of API may have different rate limits.
  Bucket identifies to which part of API the rate limits apply.
  """
  bucket string

  """
  Total Requests
  Total requests available in the time window.
  """
  totalRequests number

  """
  Remaining Requests
  Remaining requests available in the time window.
  """
  remainingRequests number

  """
  Remaining Requests available in percents
  Remaining requests available in the time window in percents.
  """
  remainingRequestsPercentage number

  """
  Reset Timestamp
  Timestamp when the rate limits will reset (in Unix Timestamp ms format).
  """
  resetTimestamp number

  """
  Reset After
  Number of seconds until reset of rate limit
  """
  resetAfter number

  """
  Retry After
  Number of seconds until user can retry the request
  """
  retryAfter number
}
