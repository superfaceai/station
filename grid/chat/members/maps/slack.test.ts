import { getMembersTest } from './members';

getMembersTest('slack');
