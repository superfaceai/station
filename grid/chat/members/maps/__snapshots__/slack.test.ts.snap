// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat/members/slack GetMembers performs correctly 1`] = `
Ok {
  "value": Object {
    "members": Array [
      Object {
        "id": "USLACKBOT",
        "isBot": false,
        "username": "Slackbot",
      },
    ],
    "nextPage": "dXNlcjpVRjNWMVBSSFA=",
  },
}
`;

exports[`chat/members/slack GetMembers performs correctly 2`] = `
Ok {
  "value": Object {
    "members": Array [
      Object {
        "id": "UF3V1PRHP",
        "isBot": false,
        "username": "Raj",
      },
    ],
    "nextPage": "dXNlcjpVRjRQUkJITDc=",
  },
}
`;
